### PR TITLE
AWS GPU (split)

### PR DESCRIPTION
**Pull request by kwasd**
This MR adds support for running NVIDIA GPUs on AWS EKS.
However, this may broke existing GKE code before #2 is merged into main.

### CODE REVIEW

File: terraform/aws/main.tf [hunk 1]
```diff
@@ -57,16 +67,25 @@ module "eks" {
...
-      max_size = 3
-      desired_size = 3
-      instance_types = ["t3.xlarge"]
+      min_size = 2
+      max_size = 2
+      disk_size = 100
+      desired_size = 2
+      ami_type = "AL2_x86_64"
+      ami_id = data.aws_ami.current_aws_ami.id
+      #ami_id = "ami-003da7190fab29a6c"
```
Comment (pbchekin) [inline terraform/aws/main.tf:82]: Remove comment?

File: terraform/modules/ray/main.tf [hunk 2]
```diff
@@ -100,8 +100,8 @@ resource "helm_release" "ray" {
                   topologyKey: kubernetes.io/hostname
       worker:
         replicas: "${var.ray_worker_nodes}"
-        minReplicas: "${var.ray_worker_nodes}"
-        maxReplicas: "${var.ray_worker_nodes}"
+        miniReplicas: "${var.ray_worker_nodes}"
+        maxiReplicas: "${var.ray_worker_nodes}"
```
Comment (pbchekin) [inline terraform/modules/ray/main.tf:104]: This should not be in this PR.

File: terraform/modules/ray/variables.tf [hunk 3]
```diff
@@ -6,7 +6,7 @@ variable "namespace_labels" {
 
 variable "release" {
   description = "Version of KubeRay"
-  default = "1.1.0"
+  default = "0.6.0"
```
Comment (pbchekin) [inline terraform/modules/ray/variables.tf:9]: This should not be in this PR.

File: terraform/modules/ray/main.tf (hunk 2 above)
Comment (kwasd) [reply terraform/modules/ray/main.tf]: Thank you!

File: terraform/aws/main.tf [hunk 4]
```diff
@@ -57,16 +67,24 @@ module "eks" {
...
-      desired_size = 3
-      instance_types = ["t3.xlarge"]
+      min_size = 2
+      max_size = 2
+      disk_size = 100
+      desired_size = 2
+      ami_type = "AL2_x86_64"
+      ami_id = data.aws_ami.current_aws_ami.id
+      enable_bootstrap_user_data = true
+      instance_types = ["g4dn.xlarge"]
```
Comment (kwasd) [inline terraform/aws/main.tf:83]: Converted to draft again to introduce a mechanism for setting instance type for non-GPU clusters

**Review by pbchekin — APPROVED** [review]

**Pull request merged**